packages/medusa/src/loaders/helpers/routing/index.ts

```diff
@@ -455,16 +608,60 @@ export class RoutesLoader {
 
       const routes = descriptor.config.routes
 
+      /**
+       * Apply default store and admin middlewares if
+       * not opted out of.
+       */
+
       if (descriptor.config.shouldAppendCustomer) {
-        this.app.use(descriptor.route, authenticateCustomer())
+        /**
+         * Add the customer to the request object
+         */
+        this.router.use(descriptor.route, authenticateCustomer())
+      }
+
+      if (descriptor.config.shouldRequireCustomerAuth) {
+        /**
+         * Require the customer to be authenticated
+         */
+        this.router.use(descriptor.route, requireCustomerAuthentication())
+      }
+
+      if (descriptor.config.shouldAppendStoreCors) {
```

Comment:
suggestion: I would say that the cors should be applied before any other middleware, otherwise you can play with the auth and body parser from any origin no?